src/snmalloc/backend_helpers/lockrange.h

```diff
@@ -7,6 +7,9 @@ namespace snmalloc
 {
   /**
    * Protect the ParentRange with a spin lock.
+   *
+   * Accesses via the ancestor() mechanism will bypass the lock and so
+   * should be used only where the resulting data races are acceptable.
```

Comment:
I wonder if there is a way to add some API that requires something to be explicitly set to pass a lock range with an ancestor call.